packages/sync-server/src/sync/CentralSyncManager.ts

```diff
@@ -37,7 +69,115 @@ export class CentralSyncManager {
     return { tick: tock - 1, tock };
   }
 
-  async waitForPendingEdits(tick: number) {
+  async startSession(debugInfo = {}): Promise<StartSessionResult> {
+    // as a side effect of starting a new session, cause a tick on the global sync clock
+    // this is a convenient way to tick the clock, as it means that no two sync sessions will
+    // happen at the same global sync time, meaning there's no ambiguity when resolving conflicts
+
+    const sessionId = generateId();
+    const startTime = new Date();
+
+    const syncSession = await this.models.syncSession.create({
+      id: sessionId,
+      startTime,
+      lastConnectionTime: startTime,
+      debugInfo,
+    });
+
+    log.info('CentralSyncManager.startSession', {
+      sessionId: syncSession.id,
+      ...debugInfo,
+    });
+
+    return { sessionId: syncSession.id };
+  }
+
+  async connectToSession(sessionId: string) {
+    const session = await this.models.syncSession.findById(sessionId);
+
+    if (!session) {
+      throw new Error(`Sync session '${sessionId}' not found`);
+    }
+
+    const { syncSessionTimeoutMs } = this.config;
+    if (
+      syncSessionTimeoutMs &&
+      !session.errors &&
+      session.updatedAt - session.createdAt > syncSessionTimeoutMs
+    ) {
+      await session.markErrored(`Sync session ${sessionId} timed out`);
```

Comment:
Feel free to ignore and resolve silently if this is too verbose:

```suggestion
      await session.markErrored(`Sync session ${sessionId} timed out after ${syncSessionTimeoutMs} ms`);
```